pypqlx.py

```diff
@@ -73,19 +73,35 @@ def __init__(self, first, last, ndata, N, S, C, L = ''):
         self.L = L if L != "" else "--"
     
     def mode(self):
-        pass
-    
+        mode = []
+        for i in sorted(set(pdf.period)):
```

Comment:
In all changes you use pdf.SOMETHING, replace to self.SOMETHING. The pdf is a global variable and not, am internal class variable.